ogre/src/OgreLidarVisual.cc

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ */
+
+#include <ignition/common/Console.hh>
+#include "ignition/rendering/ogre/OgreDynamicLines.hh"
+#include "ignition/rendering/ogre/OgreLidarVisual.hh"
+#include "ignition/rendering/ogre/OgreScene.hh"
+#include "ignition/rendering/ogre/OgreMarker.hh"
+#include "ignition/rendering/ogre/OgreGeometry.hh"
+
+
+class ignition::rendering::OgreLidarVisualPrivate
+{
+  /// \brief Non Hitting DynamicLines Object to display
+  public: std::vector<std::shared_ptr<OgreDynamicLines>> noHitRayStrips;
+
+  /// \brief Hitting DynamicLines Object to display
+  public: std::vector<std::shared_ptr<OgreDynamicLines>> rayStrips;
+
+  /// \brief Dead Zone Geometry DynamicLines Object to display
+  public: std::vector<std::shared_ptr<OgreDynamicLines>> deadZoneRayFans;
+
+  /// \brief Lidar Ray DynamicLines Object to display
+  public: std::vector<std::shared_ptr<OgreDynamicLines>> rayLines;
+
+  /// \brief The current lidar data message
+  public: std::vector<double> lidarPoints;
+
+  /// \brief True if new message is received
+  public: bool receivedMsg;
```

Comment:
can you initialize this to false? It'll also help to use a word other than msg, e.g. `receivedData` or `receivedPoints`